api/src/main/java/io/minio/MinioClient.java

```diff
@@ -4726,6 +4925,36 @@ private String initMultipartUpload(String bucketName, String objectName, Map<Str
   }
 
 
+  /**
+   * Executes uploadPartCopy multipart upload of given bucket name, object name, upload ID and parts.
+   */
+  private String uploadPartCopy(String bucketName, String objectName,String uploadId, int partNumber,
+                                Map<String, String> headerMap)
```

Comment:
I see [PUT Object](https://docs.aws.amazon.com/AmazonS3/latest/API/RESTObjectPUT.html) and [PUT Object - Copy](https://docs.aws.amazon.com/AmazonS3/latest/API/RESTObjectCOPY.html) differs only with headers. This means existing `putObject()` can be used than duplicating.
2. Similarly [Upload Part](https://docs.aws.amazon.com/AmazonS3/latest/API/mpUploadUploadPart.html) and [Upload Part - Copy](https://docs.aws.amazon.com/AmazonS3/latest/API/mpUploadUploadPartCopy.html) differs only with headers. This means existing `putObject()` can be used than duplicating.

Reply:
`Upload Part` and `Upload Part - Copy` differs in response. In `Upload Part ` etag is a part of header while in `Upload Part - Copy` the etag is a part of body.  `Initaiate multipart` and `Complete multipart` were same so used it as is. Removed 'Put Object' .